### ISSUE TITLE
Improve trip recording

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#3247]
As a carry-over from #3235, we want to re-evaluate:

**(1**) Making al trip recording setting global (i.e. the whole trip recording plugin settins screen)

**(2)** Produce no dialog upon starting a recording and create a global setting of interval. (I agree we should create the global setting, but we **should still keep the existing dialog** where a user can opt our by checking "Remember the setting". Because this dialog exactly enables on the fly configuration once we make the interval setting profile-independent. I use it a lot with different intervals for e.g. jogging vs. hiking.)

We have not discussed further but this article http://osmand.net/features?id=trip-recording-plugin#Trip_Logging:_Some_Basics_on_Parameters_and_Filtering contains 2 more small ideas we could consider:

**(3)** In the speed settings, introduce a line item "speed>0": rather simple to introduce as a line item in the current speed filter settings: This would use the GPS chipset's motion detection (it would record only points where GPS reports a speed>0). We could implement as an additional float value of 0.00001 or something, with a proper display value saying ">0."

**(4)** For the cases where we turn of GPS between recordings, I have a feeling we might improve the recording quality by either introducing a short wait time between the first GPS fix and recording a point, or by sampling e.g. 3 consecutive points and recording only the one with the lowest getAccuracy() value.

**Comment by vshcherb** [comment]
(1) Today I mostly disagree, these settings are available with profile selection, so it is quite obvious they are not global. Second, trip recording is definitely an option which somebody would like to have profile based. For most of the users who don't go to the setting we anyway display a dialog to select an interval (which I guess is global), all other settings are "good enough".
(2) ... Is this issue valid?
(3) "Not selected" indicates it though it is not very obvious but quite
(4) sounds quite difficult and I'm missing a point of a mobile device to be that "smart" if somebody cares that much about tracks, should it be online/PC tool to do that?

After one release, I still think the current implementation is good enough. To be honest, now the settings of trip recording is very-very complicated, it might be too professional or over-complicated :) 
To be honest with so much professional settings, I feel better to have it profile based.

Please let me know if we are missing any use case for non-educated users who don't go these profile settings.

**Comment by sonora** [comment]
Just trying to fix the worst blunders we have introduced, really, like:
- What happens when you browse the map in the app during a recording to check something out (e.g. a foot track), and you change the map profile for that? Your recording parameters will change without you knowing??
- What recording parameters are valid if you start a recording in the browse map profile? (We have no interface to edit these or even  determine these...??)
- We have created a Trip recording notification in Android's Notification area, and it is visible by default. It provides a button to start a recording, but it is unpredictable for the users which parameters will govern a recording triggered like this, (and there is no dialog either)?
- We have created speed filter for recordings, but the GPS algorithm's built-in "speed>0  or not" motion detection mechanism is not among them. Instead we only offer discrete values in 1km/h spacing, which of these to heuristically chose for what purpose. is rather unclear for the user.

Some of this can at least be eased by not having profile settings ...

Regarding the improvement suggestion 4:  I have run a test, recording the identical path with Osmand twice, with our current "log the first fix" and then the continuous fix method (on the same device). See these screenshots.

![asis](https://cloud.githubusercontent.com/assets/845735/24635697/561a582e-18d5-11e7-92b0-7f3652571a34.png)
![suggested](https://cloud.githubusercontent.com/assets/845735/24635698/58f7c284-18d5-11e7-9cda-d6005bb875bb.png)

Then I looked at how fresh fixes improve in 1 second intervals (just log a track in 1 second intervals, and manually turn off/on the GPS), and it appears that the fix quality increases significantly within the first 3 seconds of every fix, that's why suggestion (4) came up.
